server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -424,6 +425,7 @@ public RepositoryData addSnapshot(
             // the new master, so we make the operation idempotent
             return this;
         }
+        final var shardGenerations = updatedShardGenerations.liveIndices();
```

Comment:
Likewise this name is ambiguous; here we actually only care about the live index IDs now, maybe it'd avoid some confusion to extract that variable instead:

```suggestion
        final var liveIndexIds = updatedShardGenerations.liveIndices().indices();
```

Reply:
Yep fair call 6d148e568b8d58e5acd181b54b75810d0a02ecda